### PR TITLE
CTX-5424: Node Mode picker functionality implementation.

### PR DESCRIPTION
**Pull request by bogdant36**
In verbose mode prompt user to enter in which mode will Node be started
execution = 1
functionExclusive = 2
functionShared = 3
Pass this information to Node as a part of environment variable "CTX_NODE_MODE"
If "functionExclusive" is selected prompt user to enter a model id which will be used
Download the provided model id when Node is started if it doesn't already exists

### CODE REVIEW

File: coretex/cli/modules/node.py [hunk 1]
```diff
@@ -105,6 +109,44 @@ def registerNode(name: str) -> str:
     return accessToken
 
 
+def selectModelId(retryCount: int = 0) -> int:
+    if retryCount >= 3:
```
Comment (dule1322) [inline coretex/cli/modules/node.py:113]: There is no need for `retryCount` for this, you can remove it. `fetchById` function already has built-in retry mechanism

File: coretex/cli/modules/node.py [hunk 2]
```diff
@@ -105,6 +109,44 @@ def registerNode(name: str) -> str:
...
+        raise RuntimeError("Failed to fetch Coretex Model. Terminating...")
+
+    modelId = clickPrompt("Specify Coretex Model ID that you want to use:", type = int)
+
+    if not isinstance(modelId, int):
```
Comment (dule1322) [inline coretex/cli/modules/node.py:118]: In `clickPrompt` type was already marked as `int`, is there a need to validate it here again?

File: coretex/cli/modules/node.py [hunk 3]
```diff
@@ -105,6 +109,44 @@ def registerNode(name: str) -> str:
...
+        "Execution": NodeMode.execution,
+        "Function exclusive": NodeMode.functionExclusive,
+        "Function shared": NodeMode.functionShared
+    }
+    choices = list(availableNodeModes.keys())
+
+    stdEcho("Please select Coretex Node mode:")
+    selectedMode = arrowPrompt(choices)
+
+    if not availableNodeModes[selectedMode] == NodeMode.functionExclusive:
```
Comment (dule1322) [inline coretex/cli/modules/node.py:143]: Let's invert this `if` statement

File: coretex/cli/modules/docker.py [hunk 4]
```diff
@@ -59,13 +59,15 @@ def start(
...
     nodeSharedMemory: int,
+    nodeMode: int
 ) -> None:
 
     runCommand = [
         "docker", "run", "-d",
         "--env", f"CTX_API_URL={serverUrl}",
         "--env", f"CTX_STORAGE_PATH={storagePath}",
         "--env", f"CTX_NODE_ACCESS_TOKEN={nodeAccessToken}",
+        "--env", f"CTX_NODE_MODE={nodeMode}",
```
Comment (dule1322) [inline coretex/cli/modules/docker.py:70]: Do we need to update the bash script for auto update with this new environment variable?

**Review by dule1322 — CHANGES_REQUESTED** [review]

File: coretex/cli/modules/node.py (hunk 1 above)
Comment (bogdant36) [reply coretex/cli/modules/node.py:113]: This retry count I've added is copying old (typescript) CLI mechanishm, so if user specifies invalid model id he will have prompt again and if same happens again he will have one more chance to specifiy good model id before terminating CLI.

**Review by dule1322 — APPROVED** [review]

File: coretex/cli/modules/user.py [hunk 5]
```diff
@@ -77,30 +77,35 @@ def initializeUserSession() -> None:
...
-            refreshToken = config["refreshToken"]
-            response = networkManager.authenticateWithRefreshToken(refreshToken)
-            if response.hasFailed():
-                if response.statusCode >= 500:
-                    raise NetworkRequestError(response, "Something went wrong, please try again later.")
-
-                if response.statusCode >= 400:
-                    response = authenticateUser(config["username"], config["password"])
+        tokenExpirationDate = config.get("tokenExpirationDate", None)
+        refreshTokenExpirationDate = config.get("refreshTokenExpirationDate", None)
```
Comment (dule1322) [inline coretex/cli/modules/user.py:81]: No need to specify `None` as default value when using `dict.get`

File: coretex/cli/modules/user.py [hunk 6]
```diff
@@ -77,30 +77,35 @@ def initializeUserSession() -> None:
...
         else:
             response = authenticateUser(config["username"], config["password"])
 
         jsonResponse = response.getJson(dict)
         config["token"] = jsonResponse["token"]
         config["tokenExpirationDate"] = jsonResponse["expires_on"]
-        config["refreshToken"] = jsonResponse.get("refresh_token", config["refreshToken"])
-        config["refreshTokenExpirationDate"] = jsonResponse.get("refresh_expires_on", config["refreshTokenExpirationDate"])
+        config["refreshToken"] = jsonResponse.get("refresh_token", None)
+        config["refreshTokenExpirationDate"] = jsonResponse.get("refresh_expires_on", None)
```
Comment (dule1322) [inline coretex/cli/modules/user.py:109]: Same here

**Review by dule1322 — CHANGES_REQUESTED** [review]

**Review by dule1322 — APPROVED** [review]

**Pull request merged**